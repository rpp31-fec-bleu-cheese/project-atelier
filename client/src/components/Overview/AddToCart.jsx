import React from 'react';
import PropTypes from 'prop-types';
import { useState, useEffect } from 'react';
import axios from 'axios';
import config from '../../../../config.js';

// Add to Cart Component
let AddToCart = ({productStyles, indexes, changeInOutfit, outfitIds}) => {

  let stylesIndex = indexes.style;
  const [currentSize, setCurrentSize] = useState(null);
  const [currentQuantity, setCurrentQuantity] = useState(null);
  const [qtyInStock, setQtyInStock] = useState([]);
  const [outOfStock, setOutOfStock] = useState(false);
  const [myOutfitIcon, setMyOutfitIcon] = useState('⭐');

  useEffect(() => {
    setCurrentQuantity(null);
    setCurrentSize(null);
    setQtyInStock([]);
  }, [stylesIndex]);

  let handleMyOutfitCollection = () => {
    if (outfitIds.includes(Number(productStyles.product_id))) {
      setMyOutfitIcon('❤️');
    } else {
      setMyOutfitIcon('⭐');
    }
  }

  useEffect(() => {
    handleMyOutfitCollection();
  }, [outfitIds])

  useEffect(() => {
    handleMyOutfitCollection()
  }, [productStyles.product_id])


  if(Object.keys(productStyles).length) {

    let skusArray = [];
    let currentSkus = productStyles.results[indexes.style].skus;
    for (let key in currentSkus) {
      skusArray.push(currentSkus[key]);
    }

    // this variable correctly tracks product and style id for AddToBag or MyOutfit
    let currentProductStyleData = {productId: productStyles.product_id,
      styleId: productStyles.results[indexes.style].style_id}


    let handleMyOutfitClick = (event) => {
      event.persist();
      let productNumber = productStyles.product_id;
      if(typeof productNumber === 'string') {
        productNumber = Number(productStyles.product_id);
      }
      if (myOutfitIcon === '⭐') {
        if(!outfitIds.includes(productNumber)) {
          changeInOutfit(event, productNumber, 'Add');
        }
        setMyOutfitIcon('❤️');
      }
      if (myOutfitIcon === '❤️') {
        changeInOutfit(event, productNumber, 'Delete');
        setMyOutfitIcon('⭐');
      }
      event.preventDefault();
    }




    let handleSizeClick = (event) => {
      event.persist();
      setOutOfStock(false);
      let selectedSize = event.target.value;
      for (var i = 0; i < skusArray.length; i++) {
        if (skusArray[i].size === selectedSize) {
          let availableQuantity = skusArray[i].quantity;
          let quantityArray = [];
          if (!availableQuantity) {
            setOutOfStock(true);
          } else if (availableQuantity <= 15) {
            for (let q = 1; q <= availableQuantity; q++) {
              quantityArray.push(q);
            }
          } else {
            for (let q = 1; q <= 15; q++) {
              quantityArray.push(q);
            }
          }
          setCurrentSize(selectedSize);
          setQtyInStock(quantityArray);
        }
      }

      event.preventDefault();
    }

    let handleQuantityClick = (event) => {
      event.persist();
      let selectedQty = event.target.value;
      setCurrentQuantity(selectedQty);
      event.preventDefault();
    }

    let handleAddToBag = () => {
      // get sku
      let productSkuForBag;
      for (let key in currentSkus) {
        if (currentSize === currentSkus[key].size) {
          productSkuForBag = key;
        }
      }
      let optionsForCart = {
        url: '/cart',
        method: 'post',
        data: {sku_id: productSkuForBag}
      };
      axios(optionsForCart)
        .then(response => {
          console.log('Response from Cart:', response.data);
        })
          .catch(error => {
            console.log(error);
          });
    }



    return (
      <div className="AddToCart">
        <div className="SizeSelector">
          <form>
            <select data-testid="SizeSelectorDropdown" className="SizeSelectorDropdown" onChange={handleSizeClick}>
              {!outOfStock && <option value="">SELECT SIZE</option>}
              {skusArray.map((skuData, i) => (
                <option data-testid={`SizeOption ${i}`} className="SizeOption" key={i} value={skuData.size} selected={currentSize === skuData.size ? true : false}>{skuData.size}</option>
              ))}
            </select>
          </form>
        </div>
        <div className="QuanititySelector">
          <form>
            <select data-testid="QuanititySelectorDropdown" className="QuanititySelectorDropdown" onChange={handleQuantityClick}>
            <option value="">-</option>
              {qtyInStock.length && qtyInStock.map((qty, i) => (
                <option data-testid={`QuantityOption ${i}`} className="QuantityOption" key={i} value={qty} selected={currentQuantity === qty ? true : false}>{qty}</option>
              ))}
            </select>
          </form>
        </div>
         {outOfStock && <div className="AddToBag">
          <button className="AddToBagButton" style={{color: 'red'}} alt="Product Out of Stock">OUT OF STOCK</button>
        </div>}
        {!outOfStock && <div className="AddToBag">
          <button data-testid="AddToBagButton" className="AddToBagButton" onClick={handleAddToBag} alt="Add Product to Bag">Add to Bag</button>
        </div>}
        <div className="AddToFavorite">
          <button data-testid="AddToFavoriteButton" className="AddToFavoriteButton" onClick={handleMyOutfitClick} alt="Add Product to My Outfit">{myOutfitIcon}</button>
        </div>
      </div>
    );
  } else {
    return (
      <div className="AddToCart">
        <h3>Loading...</h3>
      </div>
    );
  }
}

AddToCart.propTypes = {
  productStyles: PropTypes.object,
  indexes: PropTypes.object,
  changeInOutfit: PropTypes.func,
  outfitIds: PropTypes.array
}

export default AddToCart;
export const handleAddToBag = () => {
  // get sku
  let productSkuForBag;
  for (let key in currentSkus) {
    if (currentSize === currentSkus[key].size) {
      productSkuForBag = key;
      // console.log('Product Sku:', productSkuForBag);
    }
  }
  let optionsForCart = {
    url: '/cart',
    method: 'post',
    headers: {'Content-Type': 'application/json',
    'Authorization': config.API_KEY},
    data: {sku_id: productSkuForBag}
  };
  axios(optionsForCart)
    .then(response => {
      console.log('Response from Cart:', response.data);
    })
      .catch(error => {
        console.log(error);
      });
}