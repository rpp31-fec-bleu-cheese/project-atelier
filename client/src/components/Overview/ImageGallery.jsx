import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCameraRetro } from '@fortawesome/free-solid-svg-icons';
import { faArrowCircleLeft } from '@fortawesome/free-solid-svg-icons';
import { faArrowCircleRight } from '@fortawesome/free-solid-svg-icons';
import { faExpandArrowsAlt } from '@fortawesome/free-solid-svg-icons';
import { faChevronUp } from '@fortawesome/free-solid-svg-icons';
import { faChevronDown } from '@fortawesome/free-solid-svg-icons';


let ImageGallery = ({handleLeftArrowClick, handleRightArrowClick, handleThumbnailClick, indexes, productStyles}) => {

  const [showModal, setShowModal] = useState(false);

  let handleModalClick = () => {
    setShowModal(true);
  }

    if (Object.keys(productStyles).length) {

    return (
      <div className="ImageGallery">{
        productStyles.results[indexes.style].photos[indexes.photo].url ?
        <div data-testid="MainImage" style={{background: `center / contain no-repeat url(${productStyles.results[indexes.style].photos[indexes.photo].url})`}} className="MainImage" alt="Main Product Image">
        </div>
        :
        <div data-testid="MainImageUnavailable" className="MainImageUnavailable">
        <FontAwesomeIcon icon={faCameraRetro} />
        </div>
        }
        <div className="ImageGalleryControls">
        <button className="ThumbnailScrollUp"><FontAwesomeIcon icon={faChevronUp} alt="Scroll Gallery Up"/></button>
        <div data-testid="ImageGalleryThumbnails" className="ImageGalleryThumbnails">
        {productStyles.results[indexes.style].photos.map((currentStyle, i) => (
            indexes.photo === i ?

            <div data-testid={`GalleryThumbnail ${i}`} key={i} index={i} style={{background: `center / contain no-repeat url(${currentStyle.thumbnail_url})`}} className="GalleryThumbnailSelected" onClick={handleThumbnailClick} alt="Product Thumbnail Image"></div>
            :
            <div data-testid={`GalleryThumbnail ${i}`} key={i} index={i} style={{background: `center / contain no-repeat url(${currentStyle.thumbnail_url})`}} className="GalleryThumbnail" onClick={handleThumbnailClick} alt="Product Thumbnail Image"></div>
          ))}
        </div>
        <button className="ThumbnailScrollDown"><FontAwesomeIcon icon={faChevronDown} alt="Scroll Gallery Down"/></button>
        </div>
        <div className="SlideGalleryLeft">
          <div data-testid="SlideGalleryLeftButtonTest" className="SlideGalleryLeftButton" onClick={handleLeftArrowClick}><FontAwesomeIcon icon={faArrowCircleLeft} /></div>
        </div>
        <div className="SlideGalleryRight">
          <div data-testid="SlideGalleryRightButtonTest" className="SlideGalleryRightButton" onClick={handleRightArrowClick}><FontAwesomeIcon icon={faArrowCircleRight} /></div>
        </div>
        <div className="OpenGalleryModal">
          <div data-testid="OpenGalleryModalButton" className="OpenGalleryModalButton" onClick={handleModalClick}><FontAwesomeIcon icon={faExpandArrowsAlt} /></div>
            {showModal && <div data-testid="GalleryModal" id="GalleryModal" onClick={() => { setShowModal(false) }}>
              <div id="GalleryModalContent" className="GalleryModalContent">
                <img className="ModalImage" src={productStyles.results[indexes.style].photos[indexes.photo].url} alt="Product Image Close Up" />
              </div>
            </div>}
        </div>
      </div>
    );
  } else {
    return (
      <div className="ImageGallery">
        <h3>Loading...</h3>
      </div>
    )
  }
}

ImageGallery.propTypes = {
  products: PropTypes.array,
  handleLeftArrowClick: PropTypes.func,
  handleRightArrowClick: PropTypes.func,
  handleThumbnailClick: PropTypes.func,
  indexes: PropTypes.object,
  productStyles: PropTypes.object
}

export default ImageGallery;