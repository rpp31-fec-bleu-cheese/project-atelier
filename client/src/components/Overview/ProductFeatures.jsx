import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';


let ProductFeatures = ({productById}) => {

  return (
    <div className="ProductFeatures">
      {(Object.keys(productById).length) && <div className="ProductFeaturesList" data-testid="FeatureList">
        {productById.features.map((productFeature, i) => (
          <div data-testid={`FeatureListEntry ${i}`} key={i}>âœ” {productFeature.feature}: {productFeature.value}</div>
        ))}
      </div>}
    </div>
  );
}

ProductFeatures.propTypes = {
  productById: PropTypes.object,
  cam_token: PropTypes.object
}

export default ProductFeatures;