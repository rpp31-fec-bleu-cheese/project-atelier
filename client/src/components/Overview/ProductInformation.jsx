import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';


let ProductInformation = ({productById, productStyles, indexes, starRating}) => {

  if (Object.keys(productStyles).length) {
    return (
      <div className="ProductInformation">
        <div className="StarsAndReviews">
          <div data-testid="ProductStars">
          <StarRating starRating={starRating}/>
          </div>
          <div>
          <a href="#RatingsReviews" data-testid="ProductReviewsLink">Read all reviews</a>
          </div>
        </div>
        <div className="ProductCategory" data-testid="ProductByCategory">
          {productById.category}
        </div>
        <div className="ProductName" data-testid="ProductByName">
          {productById.name}
        </div>
        <div data-testid="ProductByPrice" className="ProductPrice">
          {!productStyles.results[indexes.style].sale_price && <div>{`$${productStyles.results[indexes.style].original_price}`}</div>}
          {productStyles.results[indexes.style].sale_price && <div><div style={{textDecoration: 'line-through'}}>{`$${productStyles.results[indexes.style].original_price}`}</div><div data-testid="ProductBySalePrice" style={{color: 'red'}}>{`$${productStyles.results[indexes.style].sale_price}`}</div></div>}
        </div>
      </div>
    );
  } else {
    return (
      <div className="ProductInformation">
        <h3>Loading...</h3>
      </div>
    );
  }
}

let StarRating = ({starRating}) => {
  return (
    <div className='StarsContainer'>
    <div className='Stars'>
      <div className='empty-stars'></div>
      <div className='full-stars' style={{width: ((starRating / 5) * 100) + '%' }}></div>
    </div>
  </div>
  )
}


ProductInformation.propTypes = {
  productById: PropTypes.object,
  productStyles: PropTypes.object,
  indexes: PropTypes.object,
  starRating: PropTypes.number
}

StarRating.propTypes = {
  starRating: PropTypes.number
}

export default ProductInformation;