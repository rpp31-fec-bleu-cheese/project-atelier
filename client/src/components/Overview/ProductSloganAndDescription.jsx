import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';


let ProductSloganAndDescription = ({productById}) => {
  return (
    <div className="ProductSloganAndDescription">
        <div className="ProductSlogan" data-testid="ProductSloganHeader">
          {productById.slogan}
        </div>
        <div className="ProductDescription" data-testid="ProductDescriptionBody">
          {productById.description}
        </div>
    </div>
  );
}

ProductSloganAndDescription.propTypes = {
  productById: PropTypes.object
}

export default ProductSloganAndDescription;