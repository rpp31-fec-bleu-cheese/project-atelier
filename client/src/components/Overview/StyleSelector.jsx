import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheckCircle } from '@fortawesome/free-solid-svg-icons';

let StyleSelector = ({productStyles, indexes, handleStyleClick}) => {

  if (Object.keys(productStyles).length) {
    return (
      <div className="StyleSelector">
        <div className="SelectedStyle">
          <div className="SelectedStyleHeader">
          Style:
          </div>
          <div className="SelectedStyleDescription" data-testid="StyleDescription">
          {productStyles.results[indexes.style].name}
          </div>
        </div>
        <div className="StyleSelectorIcons">
          {productStyles.results.map((style, i) => (
            indexes.style === i ?
                <div key={i} index={i} style={{background: `center / contain no-repeat url(${style.photos[0].thumbnail_url})`}} className="StyleIcon" data-testid={`StyleThumbnail ${i}`} alt="Small Style Image">
                <div className="StyleIconCheck" data-testid="StyleCheck">
                <FontAwesomeIcon icon={faCheckCircle}></FontAwesomeIcon>
                </div>
                </div>
                :
                <div key={i} index={i} onClick={handleStyleClick} style={{background: `center / contain no-repeat url(${style.photos[0].thumbnail_url})`}} className="StyleIcon" data-testid={`StyleThumbnail ${i}`} alt="Small Style Image"></div>

          ))}
        </div>
      </div>
    );
  } else {
    return (
    <div className="StyleSelector">
      <h3>Loading...</h3>
    </div>
    );
  }
}

StyleSelector.propTypes = {
  productStyles: PropTypes.object,
  handleStyleClick: PropTypes.func,
  indexes: PropTypes.object
}

export default StyleSelector;