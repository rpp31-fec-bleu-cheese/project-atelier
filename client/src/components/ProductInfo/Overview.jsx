import React from 'React';
import PropTypes from 'prop-types';

let Overview = ({products}) => {



    return (
      <div id='Overview'>
        <ProductInformation products={products} />
        <StyleSelector products={products} />
      </div>
    );

  };

  let ProductInformation = ({products}) => {
    return (
      <div className="ProductInformation">
        <div className="StarsAndReviews">
          <div>
          Stars here (*****)
          </div>
          <div>
          <a href="">Read all (#) reviews</a>
          </div>
        </div>
        <div className="ProductCategory">
          {products[0].category}
        </div>
        <div className="ProductName">
          {products[0].name}
        </div>
        <div>
          {`$${products[0].default_price}`}
        </div>
      </div>
    );
  }

  let StyleSelector = ({products}) => {
    return (
      <div className="StyleSelector">
        <div className="SelectedStyle">
          <div className="SelectedStyleHeader">
          Style:
          </div>
          <div className="SelectedStyleDescription">
          Selected Style
          </div>
        </div>
        <div className="StyleSelectorIcons">
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
          <div className="StyleIcon">Test</div>
        </div>
      </div>
    );
  }


StyleSelector.propTypes = {
  products: PropTypes.array
}

Overview.propTypes = {
  products: PropTypes.array
}
ProductInformation.propTypes = {
  products: PropTypes.array
}

export default Overview;