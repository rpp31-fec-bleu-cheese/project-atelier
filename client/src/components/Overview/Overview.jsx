import React, { useState, useEffect, useLayoutEffect }  from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import config from '../../../../config.js';
import ImageGallery from './ImageGallery.jsx';
import ProductInformation from './ProductInformation.jsx';
import StyleSelector from './StyleSelector.jsx';
import AddToCart from './AddToCart.jsx';
import ProductSloganAndDescription from './ProductSloganAndDescription.jsx';
import ProductFeatures from './ProductFeatures.jsx';


let Overview = ({updateDetailsAndStyles, productById, productStyles, productId, changeInOutfit, outfitIds, starRating, trackUserClicks}) => {

  const [indexes, setIndexes] = useState({product: 0, style: 0, photo: 0});
  // console.log('INDEXES:', indexes);
  // Effect for watching incoming productId from App component
  // useLayoutEffect(() => {
  //   setIndexes({...indexes, style: 0, photo: 0})
  // }, [productId]);

  // useEffect(() => {
  //   setIndexes({...indexes, style: 0, photo: 0})
  // }, [productId]);

  useEffect (() => {
    let productIdOptions = {
      url: `/products/${productId}`,
      method: 'get'
    };
    let productStylesOptions = {
      url: `/products/${productId}/styles`,
      method: 'get'
    };
    axios(productIdOptions)
      .then(response => {
        let product = {
          details: response.data
        };
        axios(productStylesOptions)
        .then(response => {
          product.styles = response.data;
          setIndexes({...indexes, style: 0, photo: 0})
          updateDetailsAndStyles(product.details, product.styles);
            })
          })
            .catch(error => {
              console.log(error)});
  }, [productId]);


  let handleLeftArrowClick = () => {
    if (indexes.photo === 0) {
      let nextIndex = productStyles.results[indexes.style].photos.length - 1;
      setIndexes({...indexes, photo: nextIndex});
    } else {
      setIndexes({...indexes, photo: indexes.photo - 1});
    }
  }

  let handleRightArrowClick = () => {
    let nextIndex = (indexes.photo + 1) % productStyles.results[indexes.style].photos.length;
    setIndexes({...indexes, photo: nextIndex});
  }

  let handleThumbnailClick = (event) => {
    event.persist();
    let indexValue = Number(event.target.attributes.index.nodeValue);
    setIndexes({...indexes, photo: indexValue});
    event.preventDefault();
  }

  let handleStyleClick = (event) => {
    event.persist();
    let styleIndex = Number(event.target.attributes.index.nodeValue);
    if (indexes.style === styleIndex) {
      return;
    }
    console.log('Style Index:', styleIndex);
    setIndexes({...indexes, style: styleIndex, photo: 0});
    event.preventDefault();
  }

  return (
    <div data-testid="Overview" id='Overview' onClick={() => trackUserClicks('Overview', event)}>
      {Object.keys(productStyles).length > 0 && <><ImageGallery indexes={indexes} handleThumbnailClick={handleThumbnailClick} handleLeftArrowClick={handleLeftArrowClick}
       handleRightArrowClick={handleRightArrowClick} productStyles={productStyles} data-testid="ImageGallery"/>
      <ProductInformation starRating={starRating} productById={productById} productStyles={productStyles} indexes={indexes} />
      <StyleSelector productStyles={productStyles} indexes={indexes} handleStyleClick={handleStyleClick}  />
      <AddToCart productStyles={productStyles} indexes={indexes} changeInOutfit={changeInOutfit} outfitIds={outfitIds} />
      <ProductSloganAndDescription productById={productById} />
      <ProductFeatures productById={productById} /></>}
    </div>
  );
};


Overview.propTypes = {
  productById: PropTypes.object,
  productStyles: PropTypes.object,
  products: PropTypes.array,
  productId: PropTypes.number,
  changeInOutfit : PropTypes.func,
  outfitIds: PropTypes.array,
  updateDetailsAndStyles: PropTypes.func,
  starRating: PropTypes.number,
  trackUserClicks: PropTypes.func
}

export default Overview;
export const handleThumbnailClick = (event) => {
  event.persist();
  let indexValue = Number(event.target.attributes.index.nodeValue);
  setIndexes({...indexes, photo: indexValue});
  event.preventDefault();
}
export const handleLeftArrowClick = () => {
  if (indexes.photo === 0) {
    let nextIndex = productStyles.results[indexes.style].photos.length - 1;
    setIndexes({...indexes, photo: nextIndex});
  } else {
    setIndexes({...indexes, photo: indexes.photo - 1});
  }
}
export const handleRightArrowClick = () => {
  let nextIndex = (indexes.photo + 1) % productStyles.results[indexes.style].photos.length;
  setIndexes({...indexes, photo: nextIndex});
}