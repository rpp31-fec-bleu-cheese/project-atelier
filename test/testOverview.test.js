// import dependencies for React, Jest, Enzyme
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, mount, render } from 'enzyme';
// import component modules to be tested
import Overview from '../client/src/components/ProductInfo/Overview.jsx';
import ImageGallery from '../client/src/components/ProductInfo/Overview.jsx';
import ProductInformation from '../client/src/components/ProductInfo/Overview.jsx';
import StyleSelector from '../client/src/components/ProductInfo/Overview.jsx';
import AddToCart from '../client/src/components/ProductInfo/Overview.jsx';
import ProductSloganAndDescription from '../client/src/components/ProductInfo/Overview.jsx';
import ProductFeatures from '../client/src/components/ProductInfo/Overview.jsx';

// configure Enzyme
configure({ adapter: new Adapter() });

// tests

describe('<Overview />', () => {
  it('Renders Overview component to the DOM', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('#Overview')).not.toBeUndefined();
    // expect(wrapper.find('#Overview')).to.have.lengthOf(6);
  })
  it('Contains an Image Gallery component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('ImageGallery')).not.toBeUndefined();
  })
  it('Contains a Product Information component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('ProductInformation')).not.toBeUndefined();
  })
  it('Contains a Style Selector component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('StyleSelector')).not.toBeUndefined();
  })
  it('Contains a Add To Cart component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('AddToCart')).not.toBeUndefined();
  })
  it('Contains a Product Slogan and Description component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('ProductSloganAndDescription')).not.toBeUndefined();
  })
  it('Contains a Product Features component', () => {
    const wrapper = shallow(<Overview />);
    expect(wrapper.find('ProductFeatures')).not.toBeUndefined();
  })



  // it('Renders child components', () => {
  //   const wrapper = shallow((
  //     <Overview>
  //       <ImageGallery />
  //     </Overview>
  //   ));
  //   expect(wrapper.find(<ImageGallery />)).to.equal(true);
  // });


})

describe('<ImageGallery />', () => {
  it('Renders an Image Gallery container', () => {
    const wrapper = shallow(<ImageGallery />);
    expect(wrapper.find('.ImageGallery')).not.toBeUndefined();
  })

  it('Renders images in Gallery Thumbnails container', () => {
    const wrapper = shallow(<ImageGallery />);
    expect(wrapper.find('.ImageGalleryThumbnails').children()).not.toBeUndefined();
  })
})

describe('<ProductInformation />', () => {
  it('Renders a Style Selector container', () => {
    const wrapper = shallow(<ProductInformation />);
    expect(wrapper.find('.ProductInformation')).not.toBeUndefined();
  })
})

describe('<StyleSelector />', () => {
  it('Renders a Style Selector container', () => {
    const wrapper = shallow(<StyleSelector />);
    expect(wrapper.find('.StyleSelector')).not.toBeUndefined();
  })
  it('Renders styles in Style Selector Icons container', () => {
    const wrapper = shallow(<StyleSelector />);
    expect(wrapper.find('.StyleSelectorIcons').children()).not.toBeUndefined();
  })
})

describe('<AddToCart />', () => {
  it('Renders an Add to Cart container', () => {
    const wrapper = shallow(<AddToCart />);
    expect(wrapper.find('.AddToCart')).not.toBeUndefined();
  })
  it('Renders sizes in Size Selector Dropdown container', () => {
    const wrapper = shallow(<AddToCart />);
    expect(wrapper.find('.SizeSelectorDropdown').children()).not.toBeUndefined();
  })
  it('Allows items to be added to bag', () => {
    const wrapper = mount(<AddToCart />);
    const button = wrapper.find('.AddToBagButton').text();
    expect(button).toEqual('Add to Bag');
  })
  // it('Allows a size to be selected', () => {
  //   const wrapper = shallow(<AddToCart />);
  //   wrapper.find('.SizeOption').simulate('click');
  //   expect(handleSizeClick).to.have.property('callCount', 1);
  // })
})

describe('<ProductSloganAndDescription />', () => {
  it('Renders a Product Slogan and Description container', () => {
    const wrapper = shallow(<ProductSloganAndDescription />);
    expect(wrapper.find('.ProductSloganAndDescription')).not.toBeUndefined();
  })
})

describe('<ProductFeatures />', () => {
  it('Renders a Product Features container', () => {
    const wrapper = shallow(<ProductFeatures />);
    expect(wrapper.find('.ProductFeatures')).not.toBeUndefined();
  })
})


// it('Renders six child components', () => {
//   const wrapper = shallow(<Overview />);
//   expect(wrapper.find(ImageGallery)).to.have.lengthOf(1);
// })