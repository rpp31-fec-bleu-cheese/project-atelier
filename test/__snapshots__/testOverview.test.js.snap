// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddToCart /> Allows items to be added to bag 1`] = `
<div
  data-testid="Overview"
  id="Overview"
/>
`;
